keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,545 @@
+  interact with it. They think deleting the ConfigMap would delete the local
+  registry (which it does not).
+
+- This KEP does not specify any automatic mechanism for keeping the
+  RegistryStatus up-to-date with how the cluster is configured.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+
+Tools that configure a local registry should apply the following ConfigMap to
+the cluster:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: registry
+  namespace: kube-public
+data:
+  RegistryStatus: |
+    apiVersion: registry.x-k8s.io/v1alpha1
+    kind: RegistryStatus
+    host: "localhost:5000"
+    hostFromCluster: "registry:5000"
+    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
+```
+
+Documentation that educates people on how to set up a registry manually should
+include instructions on this.
+
+The name of the config map must be `registry`.
+
+The namespace must be `kube-public`.
+
+The `RegistryStatus` field of the ConfigMap data takes the form of a component
+config, like the `kubeadm-config` ConfigMap in `kube-system`.
+
+There are 3 fields:
+
+- `host`: Advertises the host (hostname and port) of the registry, as seen from outside the cluster. 
+  This is the registry host that tools should push images to.
+  
+- `hostFromCluster`: Advertises the host (hostname and port) of the registry, as
+  seen from the cluster's container runtime. This is the registry that should
+  appear in Kubernetes objects applied to the cluster. Should be ignored if
+  `host` is not set.  If omitted, assumed to be the same as `host`.
+  
+- `help`: Adverises help documentation for users that want to set up a
+   registry. Tools can use this to nudge users to enable the registry.
+   
+I don't have a strong sense of whether `registry.x-k8s.io/v1alpha1` is the right value
+for `apiVersion`, or if there's guidance on a better convention.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).  Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+Not applicable.
```

Comment:
i guess for a versioned component config, unit tests for the API machinery are a minimum required in the test plan.

Reply:
I added some comments on this